libbeat/mapping/field.go

```diff
@@ -119,6 +122,55 @@ func (f *Field) Validate() error {
 	return nil
 }
 
+func (f *Field) validateType() error {
+	switch strings.ToLower(f.Type) {
+	case "text", "keyword":
+		return errors.Wrapf(validateFormat(f.Format, "string"), "field %s", f.Name)
+	case "long", "integer", "short", "byte", "double", "float", "half_float", "scaled_float":
+		return errors.Wrapf(validateFormat(f.Format, "number"), "field %s", f.Name)
+	case "date", "date_nanos":
+		return errors.Wrapf(validateFormat(f.Format, "date"), "field %s", f.Name)
+	case "geo_point":
+		return errors.Wrapf(validateFormat(f.Format, "geo_point"), "field %s", f.Name)
+	case "boolean", "binary", "ip", "alias", "array":
+		if f.Format != "" {
+			return fmt.Errorf("no format expected for field %s, found: %s", f.Name, f.Format)
+		}
+	case "object", "group", "nested":
+		// No check for them yet
+	case "":
+		// Module keys, not used as fields
+	default:
+		// There are more types, not being used by beats, to be added if needed
+		return fmt.Errorf("unexpected type '%s' for field '%s'", f.Type, f.Name)
+	}
+	return nil
+}
+
+var validFormats = map[string][]string{
+	"string":    []string{"string", "url"},
+	"number":    []string{"string", "url", "bytes", "duration", "number", "percent", "color"},
+	"date":      []string{"string", "url", "date"},
+	"geo_point": []string{"geo_point"},
+	"none":      []string{},
+}
+
+func validateFormat(format, group string) error {
+	if format == "" {
+		return nil
+	}
+	groupFormats, ok := validFormats[group]
+	if !ok {
+		panic("unexpected type group " + group)
```

Comment:
We can return an error, is a `panic` necessary?

Reply:
This is a function only used internally, where `group` can only be set in `validateType`, so an incorrect value here is probably a bug.
In any case, I have modified it so a type group has its own type and so it can be checked by the compiller, that is its job :slightly_smiling_face: 
Let me know what you think.